### PR TITLE
rgw : fix the inheriting class member is passed to base ctor

### PR DESCRIPTION
**Pull request by BryceCao**
Fixes: https://tracker.ceph.com/issues/57324
Signed-off-by: caolei <cao.leilc@inspur.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/store/rados/rgw_bucket.cc [hunk 1]
```diff
@@ -2066,7 +2066,7 @@ class RGWMetadataHandlerPut_BucketInstance : public RGWMetadataHandlerPut_SObj
                                        RGWSI_MetaBackend_Handler::Op *_op, string& entry,
                                        RGWMetadataObject *_obj, RGWObjVersionTracker& objv_tracker,
 				       optional_yield y,
-                                       RGWMDLogSyncType type, bool from_remote_zone) : RGWMetadataHandlerPut_SObj(_handler, _op, entry, obj, objv_tracker, y, type, from_remote_zone),
+                                       RGWMDLogSyncType type, bool from_remote_zone) : RGWMetadataHandlerPut_SObj(_handler, _op, entry, nullptr, objv_tracker, y, type, from_remote_zone),
```
Comment (yuvalif) [inline src/rgw/store/rados/rgw_bucket.cc:2069]: why do you pass null?
isn't the correct value should be the one coming from `_obj`?

Comment (BryceCao) [reply src/rgw/store/rados/rgw_bucket.cc]: has been changed to `_obj`. PTAL

> why do you pass null? isn't the correct value should be the one coming from `_obj`?

**Review by yuvalif — APPROVED** [review]

**Comment by cbodley** [comment]
passed qa in https://pulpito.ceph.com/cbodley-2022-12-20_15:43:17-rgw-wip-cbodley-testing-distro-default-smithi/

**Comment by yuvalif** [comment]
> passed qa in https://pulpito.ceph.com/cbodley-2022-12-20_15:43:17-rgw-wip-cbodley-testing-distro-default-smithi/

@cbodley is this crash: http://qa-proxy.ceph.com/teuthology/cbodley-2022-12-20_15:43:17-rgw-wip-cbodley-testing-distro-default-smithi/7122631/remote/smithi006/crash/2022-12-20T17%3A06%3A15.418679Z_56c96da9-cf2c-4ed5-8bcd-2b605993a7f8/
expected?

**Pull request merged**